File: JPAHospitalReview/src/main/java/entities/Post.java
package entities;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;
import javax.persistence.Table;

@Entity
@Table(name="post")
public class Post {

	// F I E L D S
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name="id")
	private int id;
	
	@Column(name="review")
	private String review;
	
	@Column(name="create_date")
	private String createDate;
	
	@OneToOne
	@JoinColumn(name="user_id")
	private User user;
	
	@OneToOne
	@JoinColumn(name="hospital_id")
	private Hospital hospital;
	
	@OneToOne
	@JoinColumn(name="hospital_id")
	private Procedure procedure;
	
	// C O N S T R U C T O R S
	public Post() {
		super();
	}
}

File: ngHospitalReview/src/app/models/post.ts
import { Procedure } from './procedure';
import { Hospital } from './hospital';
import { User } from './user';


export class Post {
  // F I E L D S
  id: number;
  review: string;
  createDate: string;
  user: User;
  hospital: Hospital;
  procedure: Procedure;

  // C O N S T R U C T O R
  constructor(id?: number, review?: string, createDate?: string, user?: User, hospital?: Hospital, procedure?: Procedure){
    this.id = id;
    this.review = review;
    this.createDate = createDate;
    this.user = user;
    this.hospital = hospital;
    this.procedure = procedure;
  }
}
